core/src/Kokkos_Tuners.hpp

```diff
@@ -523,7 +525,7 @@ class TeamSizeTuner : public ExtendableTunerMixin<TeamSizeTuner> {
   }
 
   template <typename... Properties>
-  void tune(Kokkos::TeamPolicy<Properties...>& policy) {
+  auto tune(Kokkos::TeamPolicy<Properties...>& policy) {
```

Comment:
This now returns its result after mutating it, to make it compatible both with the old version and the new generic one. Arguably, we could make this not mutate anymore, but _technically_ that's a breaking change

Reply:
I'll resolve this, as I'm now using that model. Crefs in, new policies out